tests/test_page.py

```diff
@@ -46,6 +68,12 @@ def test_wait_for_page_timeout(page, selenium):
         page.wait_for_page_to_load()
 
 
+def test_wait_for_page_empty_base_url(page, selenium):
+    selenium.current_url = str(random.random())
+    page.base_url = None
```

Comment:
Rather than setting the `base_url` to `None` could we instantiate a new `Page` without the argument? That would more closely reflect actual use of the package.